src/seedu/addressbook/data/person/Address.java

```diff
@@ -37,22 +55,53 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return block + ", " + street + ", " + unit + ", " + postalCode;
     }
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+        if (other == this) // short circuit if same object
+            return true;
+        
+        if (other instanceof Address) { // instanceof handles nulls
+            Address otherAddress = (Address)other;
+            return otherAddress.block.equals(block) &&
```

Comment:
conventionally, don't you say `this.block.equals(other)` rather than what you have used? just a thought..